src/rust/catmem/mod.rs

```diff
@@ -121,10 +141,9 @@ impl CatmemLibOS {
     /// Closes a memory queue.
     pub fn close(&mut self, qd: QDesc) -> Result<(), Fail> {
         trace!("close() qd={:?}", qd);
-        match self.rings.remove(&qd) {
-            Some(pipe) => {
-                self.push_eof(pipe.buffer())?;
-                self.qtable.free(qd);
+        match self.qtable.free(&qd) {
+            Some(queue) => {
+                self.push_eof(queue.get_pipe().buffer())?;
```

Comment:
The new code is now removing the queue descriptor from the queue table, even if we fail to push the EOF.

Reply:
fixed this.